### PR TITLE
[11912] [skip ci] Add how-to Guide for Azure operators 

### LINKED ISSUE #11912: Add how-to Guide for Azure operators
**Description**

A guide that describes how to use all the operators for Azure (https://github.com/apache/airflow/tree/master/airflow/providers/microsoft/azure) would be useful.

Other guides are available:
https://airflow.readthedocs.io/en/latest/howto/operator/index.html

Source code for those guides are at:
https://github.com/apache/airflow/tree/master/docs/howto/operator

Are you wondering how to start contributing to this project? Start by reading our [contributor guide](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst)

Best regards,
Kaxil

**Related Issues**

Missing guides for operators in Microsoft provider:
- MSSQL Operators: https://github.com/apache/airflow/issues/11913
- WinRM: https://github.com/apache/airflow/issues/11914

### PR DESCRIPTION
**Pull request by pyerbiz**
Adding documentation and example dags for missing cases in Azure Providers.

closes: #11912
related:

### CODE REVIEW

File: airflow/providers/microsoft/azure/example_dags/example_adx_query.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# :param query: KQL query to run (templated).
+# :type query: str
+# :param database: Database to run the query on (templated).
+# :type database: str
+# :param options: Optional query options. See:
+#     https://docs.microsoft.com/en-us/azure/kusto/api/netfx/request-properties#list-of-clientrequestproperties
+# :type options: dict
+# :param azure_data_explorer_conn_id: Reference to the
+#     :ref:`Azure Data Explorer connection<howto/connection:adx>`.
+# :type azure_data_explorer_conn_id: str
```
Comment (ephraimbuddy) [inline airflow/providers/microsoft/azure/example_dags/example_adx_query.py:36]: Maintaining the above parameters here would be problematic. For example, if there's an addition to the params, we may forget to update the param list here

**Review by ephraimbuddy — CHANGES_REQUESTED** [review]
I think we should make each example have a system test too; that way, we can confirm that the example works.

Here's an example system test: https://github.com/apache/airflow/blob/main/tests/providers/microsoft/azure/operators/test_adls_delete_system.py

So I suggest we take this, one example and a system test at a time

**Comment by pyerbiz** [comment]
I am back from a work and travel hiatus. If still acceptable, maintainers may re-assign this issue to me.

**Comment by uranusjr** [comment]
You’re still assigned in the issue, please feel free to continue.

**Comment by kaxil** [comment]
@pyerbiz Do you need any help with this issue

**Comment by pyerbiz** [comment]
@kaxil Not right now. I'll try to this weekend and seek help if I need it.

**Comment by pyerbiz** [comment]
@kaxil I had a question about the example_dag for [adx.py](https://github.com/apache/airflow/blob/main/airflow/providers/microsoft/azure/operators/adx.py).

I am using gcp's big query docs for reference and also the postgres operator doc. Based on them, to make a complete example I should be able to create table, load data, query data, and drop table. I have to tried to do that in kusto above. I am concerned about the connection_string_id used to connect to ADX cluster. When I access adx with python in my local machine, I have to do the browser pop-up method, credential based login -- other methods have caused me problems in the past. I'm going to try using the AAD token method again and see if that works without pop-ups. Is there a preferred way users do adx authentication on airflow? I ask because the other example_dag files for azure all use a default "default_connection_id" string to show the example.

**Pull request closed without merging**